### PR TITLE
CE-3675 Track impressions AND viewability of the banner

### PR DESCRIPTION
**Pull request by adamkarminski**
@Wikia/spitfires 

The solution is based on Google's definition of viewability - at least 50% of an element should be visible for at least 1s during a page view.

I had to make an adjustment due to our fixed nav bar and in the line 121 I do not subtract 1/2 of banner's height since it is covered by the nav.

### CODE REVIEW

File: extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js [hunk 1]
```diff
@@ -83,14 +86,54 @@ require([
...
+		if (viewabilityCounter === 1000) {
+			clearInterval(viewabilityInterval);
+			$(w).off('scroll.trackPMPEntryPoint');
+			trackBannerViewable();
+		}
+	}
+
+	function isEntryPointInViewport() {
+		return bannerOffset > w.scrollY &&
+			(bannerBottomOffset - bannerHeight50p) < (w.scrollY + w.innerHeight);
```
Comment (adamkarminski) [inline extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js]: I will simplify this one.

File: extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js [hunk 2]
```diff
@@ -81,16 +83,60 @@ require([
...
+			clearInterval(viewabilityInterval);
+			$(w).off('scroll.trackPMPEntryPoint');
+			trackBannerViewable();
+		}
+	}
+
+	function trackBannerImpression() {
+		track({
+			label: 'first-view',
+			action: tracker.ACTIONS.VIEW
```
Comment (Grunny) [inline extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js]: Should this be `tracker.ACTIONS.IMPRESSION`?

File: extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js [hunk 3]
```diff
@@ -81,16 +83,60 @@ require([
...
+				trackBannerImpression();
+			}
+			viewabilityInterval = setInterval(calculateViewability, 10);
+		} else if (viewabilityInterval !== undefined) {
+			clearInterval(viewabilityInterval);
+		}
+	}
+
+	function calculateViewability() {
+		viewabilityCounter = viewabilityCounter + 10;
```
Comment (Grunny) [inline extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js]: `viewabilityCounter += 10;`

**Comment by Grunny** [comment]
Other than the two minor comments, LGTM.

**Comment by Grunny** [comment]
:+1:

**Pull request merged**